pkg/broker/broker.go

```diff
@@ -68,7 +62,18 @@ func (b BlockheadBroker) Provision(ctx context.Context, instanceID string, detai
 	logger := b.logger.Session("provision")
 	logger.Info("started")
 	defer logger.Info("finished")
-	return provision(logger, b.client, b.state, details)
+
+	service := b.state.Services[details.ServiceID]
+	if service == nil {
+		return brokerapi.ProvisionedServiceSpec{}, errors.New("service not found")
+	}
+
+	plan := service.Plans[details.PlanID]
+	if plan == nil {
+		return brokerapi.ProvisionedServiceSpec{}, errors.New("plan not found")
+	}
+
+	return b.manager.Provision(service, plan)
```

Comment:
I am not sure where this should be, but should we store the container id & uuid that are generated. That way we will accurately be able to check in the future if the service was actually provisioned and we need the container id so we can interact with it later.

Reply:
We also need to save the details of the port mappings so that we can send them back later.

Reply:
I think initially we were thinking of keeping everything in memory for now. Not sure if using the state struct is a good idea or whether it would make the struct overly bloated, but lets keep it in memory for now (somehow) and in a later revision we can write stuff to a db.

Reply:
I was thinking the State Struct would be a good place too. Other option could be in the broker since it's in memory, but I like the State Struct better so the broker stays as just a handler.

Reply:
As long as the state thing is interim. We need a 'persistence' layer at some point. Figure out basic files or sqlite or something. 
do-it-later